### PR TITLE
Prevent queries without at least one scalar aggregator & allow disabling this restriction

### PR DESCRIPTION
**Pull request by philkjacobs**
* updated the reloading config `jsonData` to add another field to it -- `enforceScalarSetting`. Documented this in the config UI.
* Added an allowlist for scalar aggregator names -- I don't love this solution, but didn't have anything cleaner that doesn't involve cascading changes to every single `Aggregator` class. Included the `grep` command used to generate this.
* Added `overrideScalar` field to `KairosDBTarget`, updated codepaths that create and copy these to ensure they would maintain the field and default to `false`
* Added checkbox 2-way bound to this `KairosDBTarget` field alongside copy explaining its usage.
* Updated datasource.ts to register and propagate the new config setting on datasource initialization.
* Updated `TargetValidator` to return a boolean + reason instead of just a boolean. This means we can now include customized error messages for failed validations. We can extend this to resolve similar problems/validations we want to error out for.
* Updated existing `TargetValidator` validations to have an error message.

enforceScalarAggOSSScreenshot

### CODE REVIEW

File: dist/beans/aggregators/aggregators.ts [hunk 1]
```diff
@@ -10,37 +10,54 @@ import {ScaleAggregator} from "./scale_aggregator";
...
     new RangeAggregator("min"),
-    new RangeAggregator("merge"),
     new RangeAggregator("movingWindow"),
-    new PercentileAggregator(),
-    new ApdexAggregator(),
-    new SmaAggregator(),
     new RangeAggregator("sum"),
+];
+
+export const AGGREGATORS = RANGE_AGGREGATORS.concat([
```
Comment (vjkoskela) [inline dist/beans/aggregators/aggregators.ts:28]: If an aggregator is not in the `RANGE_AGGREGATORS` list does that mean it is not considered to return a scalar? Or perhaps you could elaborate on the what these two categorizations mean (via a comment)

File: dist/beans/aggregators/aggregators.ts [hunk 2]
```diff
@@ -10,37 +10,54 @@ import {ScaleAggregator} from "./scale_aggregator";
...
-    new FilterAggregator(),
-    new Aggregator("percent_remaining")
-].sort( (a, b) => a.name.localeCompare(b.name));
+]).sort( (a, b) => a.name.localeCompare(b.name));
 
-const RANGE_AGGREGATORS = ["avg", "dev", "count", "first", "gaps",
-    "last", "least_squares", "max", "min", "gaps", "merge", "sum", "movingWindow"];
+// Loosely generated from:
+// tslint:disable-next-line
+// grep -H -q "doubleDataPointFactory" <path-to-kairosdb-extensions>/kairosdb-extensions/src/main/java/io/inscopemetrics/kairosdb/aggregators
```
Comment (vjkoskela) [inline dist/beans/aggregators/aggregators.ts:44]: This limits the list of aggregators to those supported by histograms. The Kairosdb-datasource is actually independent of histograms. You should probably also look at the aggregators in the main package as well:

https://github.com/kairosdb/kairosdb

**Comment by vjkoskela** [comment]
First, to be clear, this change is _backwards compatible_ for user behavior because it is behind a feature flag which defaults to `false`.

Second, the goal of this change (when the feature is enabled) is to prevent user's from making queries without an aggregator. There are a few specialized cases where this is necessary, for example, when visualizing a heat map. However, at this time the intent cannot be inferred from the plugin.

Instead when enabled the feature requires a user to "sign off" on user an aggregator-less query. The Metrics Portal proxy will inject a merge aggregator when configured to do so. Switching both (the MP behavior and this KDB-DS behavior) to default is something that can be discussed separately.

@philkjacobs can you provide some screenshots?

Also, it's typically bad form to link to private resources in a public pull request.

File: dist/beans/aggregators/aggregators.ts (hunk 1 above)
Comment (philkjacobs) [reply dist/beans/aggregators/aggregators.ts:28]: They're unrelated sets. I didn't think entirely reworking the Aggregator datamodel is a change worth making here. 

We could register each of these properties as a boolean (scalar, range) for every Aggregator, but as-is, some of these aggregators are little more than a name string.

I think this change is one of the ones that has not yet been merged (and isn't part of my change).

**Comment by philkjacobs** [comment]
> @philkjacobs can you provide some screenshots?
Added one.
> Also, it's typically bad form to link to private resources in a public pull request.
Fixed -- I hadn't noticed it when copying the diff summary over.

File: dist/beans/aggregators/aggregators.ts (hunk 1 above)
Comment (philkjacobs) [reply dist/beans/aggregators/aggregators.ts]: Resolving as I've removed these unrelated changes.

**Pull request merged**